x-pack/plugins/observability_solution/synthetics/server/alert_rules/status_rule/monitor_status_rule.ts

```diff
@@ -104,59 +80,103 @@ export const registerSyntheticsStatusCheckRule = (
 
-      Object.entries(downConfigs).forEach(([idWithLocation, { ping, configId }]) => {
-        const locationId = ping.observer.name ?? '';
-        const alertId = idWithLocation;
-        const monitorSummary = getMonitorSummary(
-          ping,
-          DOWN_LABEL,
-          locationId,
-          configId,
-          dateFormat,
-          tz
-        );
+      const { isLocationBased, downThreshold, numberOfLocations } = getConditionType(
+        params.condition
+      );
 
-        const { uuid, start } = alertsClient.report({
-          id: alertId,
-          actionGroup: MONITOR_STATUS.id,
+      const getConfigsByIds = () => {
+        const downConfigsById = new Map<string, AlertStatusMetaDataCodec[]>();
+        Object.entries(downConfigs).forEach(([_, config]) => {
+          const { configId } = config;
+          if (!downConfigsById.has(configId)) {
+            downConfigsById.set(configId, []);
+          }
+          downConfigsById.get(configId)?.push(config);
         });
-        const errorStartedAt = start ?? startedAt.toISOString();
+        return downConfigsById;
+      };
 
-        let relativeViewInAppUrl = '';
-        if (monitorSummary.stateId) {
-          relativeViewInAppUrl = getRelativeViewInAppUrl({
-            configId,
-            stateId: monitorSummary.stateId,
-            locationId,
-          });
+      if (isLocationBased) {
+        // for location based we need to make sure ,monitor is down for the threshold for all locations
+        // lets build a map of monitors for each location
+        const downConfigsById = getConfigsByIds();
+        for (const [configId, configs] of downConfigsById) {
+          const totalLocations = monitorLocationsMap[configId]?.length ?? 0;
+          const locationThreshold = numberOfLocations;
+          const matchingLocations = configs.length;
+          if (matchingLocations >= locationThreshold) {
+            // make sure all locations are matching down threshold as well
+            const matchingLocationsWithDownThreshold = configs.filter(
+              (config) => config.checks.down >= downThreshold
+            );
+            if (matchingLocationsWithDownThreshold.length >= locationThreshold) {
+              const monitorSummary = statusRule.getLocationBasedDownSummary({
+                statusConfigs: configs,
+                downThreshold: locationThreshold,
+                percent: numberOfLocations,
+              });
+              const alertId = `${configId}_locations_based`;
+              statusRule.scheduleAlert({
+                idWithLocation: `${configId}-${configs[0].locationId}`,
+                alertId,
+                monitorSummary,
+                statusConfig: configs[0],
+                downThreshold: locationThreshold,
+              });
+            }
+          }
         }
+      } else {
+        const groupByLocation = Boolean(params.condition?.groupByLocation ?? true);
+        if (groupByLocation) {
```

Comment:
Can we extract some of this logic into some helper functions. The nested conditionals are hard to reason about.